dask/array/core.py

```diff
@@ -1307,6 +1307,10 @@ def ravel(self):
 
     flatten = ravel
 
+    @wraps(np.choose)
+    def choose(self, choices):
+        return elemwise(variadic_choose, self, *choices)
```

Comment:
Personal preference, but I'd rather have the method call the function (smaller diff too). Not a big deal though, up to you.